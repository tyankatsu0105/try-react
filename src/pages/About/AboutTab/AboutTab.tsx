import React, { FunctionComponent } from 'react';
import { Link, Route } from 'react-router-dom';

import { AboutTabItem } from './Components/AboutTabItem';

const data = [
  {
    id: 1,
    name: '太郎',
  },
  {
    id: 2,
    name: '花子',
  },
  {
    id: 3,
    name: 'ジュニア',
  },
];

const AboutTab: FunctionComponent = () => {
  return (
    <>
      {data.map((item) => (
        <Link key={item.id} to={`/about/${item.id}`}>
          {item.name}
        </Link>
      ))}
      <Route path="/about/:userId" component={AboutTabItem} />
    </>
  );
};

export default AboutTab;
