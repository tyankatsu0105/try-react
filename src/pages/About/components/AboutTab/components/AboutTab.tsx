import React, { FunctionComponent } from 'react';
import { Link, Route } from 'react-router-dom';

import { AboutTabItem } from '../components/AboutTabItem';

export interface Props {
  data: {
    id: number;
    name: string;
    age: number;
  }[];
}

export const AboutTab: FunctionComponent<Props> = (props) => {
  const { data } = props;

  return (
    <>
      {data.map((item) => (
        <Link key={item.id} to={`/about/${item.id}`}>
          {item.name}
        </Link>
      ))}
      <Route path="/about/:userId" component={AboutTabItem} />
    </>
  );
};
