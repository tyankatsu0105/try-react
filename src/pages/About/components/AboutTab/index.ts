import { lazy } from 'react';

export const AboutTab = lazy(() => import('./containers/AboutTab'));
