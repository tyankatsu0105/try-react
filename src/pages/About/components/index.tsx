import React, { FC } from 'react';
// import spinach from './assets/images/spinach.png';

export const About: FC = () => {
  return (
    <>
      <p>
        <img src="./assets/images/spinach.png" alt="" />
      </p>
    </>
  );
};
