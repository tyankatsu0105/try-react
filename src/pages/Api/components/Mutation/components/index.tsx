import React, { FunctionComponent, useState, ChangeEvent } from 'react';
import {
  CreateIssueFunction,
  AddStarFunction,
  RemoveStarFunction,
} from '@modules/Mutation';

import { CreateIssue } from '@api/graphql/queries/pages/Api/components/__generated__/CreateIssue';
import { AddStar } from '@api/graphql/queries/pages/Api/components/__generated__/AddStar';
import { RemoveStar } from '@api/graphql/queries/pages/Api/components/__generated__/RemoveStar';

type MutationProps = {
  createIssue: CreateIssueFunction;
  addStar: AddStarFunction;
  removeStar: RemoveStarFunction;
  createIssueData: CreateIssue;
  addStarData: AddStar;
  removeStarData: RemoveStar;
};

export const Mutation: FunctionComponent<MutationProps> = (props) => {
  const {
    createIssue,
    addStar,
    removeStar,
    createIssueData,
    addStarData,
    removeStarData,
  } = props;

  const [form, setForm] = useState({
    title: '',
    body: '',
  });

  const handleInputForm = (event: ChangeEvent<HTMLInputElement>) => {
    setForm({
      ...form,
      [event.target.name]: event.target.value,
    });
  };

  return (
    <>
      {props && (
        <>
          <hr />
          <p>issue作る</p>
          <form action="">
            <input
              type="text"
              placeholder="issueタイトル"
              onChange={handleInputForm}
              value={form.title}
              name="title"
            />
            <input
              type="text"
              placeholder="本文"
              onChange={handleInputForm}
              value={form.body}
              name="body"
            />
            <button
              type="button"
              onClick={() =>
                createIssue({
                  variables: {
                    repositoryId: process.env.REACT_APP_REPOSITORY_ID,
                    title: form.title,
                    body: form.body,
                  },
                })
              }
            >
              issue作成
            </button>
          </form>

          {createIssueData &&
            createIssueData.createIssue &&
            String(JSON.stringify(createIssueData.createIssue, null, 2))}

          <hr />
          <a
            href="https://github.com/tyankatsu0105/try-react"
            target="noopener noreferrer"
            rel="noopener"
          >
            ここにstarが反映されます
          </a>
          <button type="button" onClick={() => addStar()}>
            star付与
          </button>
          {addStarData &&
            addStarData.addStar &&
            String(addStarData.addStar.clientMutationId)}
          <button type="button" onClick={() => removeStar()}>
            star削除
          </button>
          {removeStarData &&
            removeStarData.removeStar &&
            String(removeStarData.removeStar.clientMutationId)}
        </>
      )}
    </>
  );
};
