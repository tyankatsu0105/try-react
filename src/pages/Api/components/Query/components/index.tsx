import React, { FunctionComponent } from 'react';
import { GetTyankatsuRepository } from '@api/graphql/queries/pages/__generated__/GetTyankatsuRepository';

export const Query: FunctionComponent<GetTyankatsuRepository> = ({
  user,
}: GetTyankatsuRepository) => {
  return (
    <>
      {user && user.repository && (
        <>
          <ul>
            <li>{user.repository.name}</li>
            <li>{user.repository.url}</li>
            <li>{String(user.repository.viewerHasStarred)}</li>
          </ul>
        </>
      )}
    </>
  );
};
