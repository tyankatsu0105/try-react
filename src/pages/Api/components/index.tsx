import React, { FC } from 'react';
import gqlApi from '@api/graphql/queries/pages/Api.graphql';
import { useQuery } from '@apollo/react-hooks';
import { ApiQuery_organization_repositories_nodes } from '@api/graphql/queries/pages/__generated__/ApiQuery';

const Api: FC = () => {
  const { loading, data } = useQuery(gqlApi);

  return (
    <>
      {loading ? (
        <p>Loading ......</p>
      ) : (
        <ul>
          {data.organization.repositories.nodes.map(
            (node: ApiQuery_organization_repositories_nodes, index: number) => (
              <li key={node.id}>
                <h1>{index + 1}</h1>
                <a href={node.url}>{node.name}</a>
              </li>
            )
          )}
        </ul>
      )}
    </>
  );
};

export default Api;
