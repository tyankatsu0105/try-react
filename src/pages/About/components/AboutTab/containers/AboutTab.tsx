import React, { FunctionComponent } from 'react';

import { AboutTab as AboutTabComponent } from '../components/AboutTab';

const data = [
  {
    id: 1,
    name: '太郎',
    age: 25,
  },
  {
    id: 2,
    name: '花子',
    age: 20,
  },
  {
    id: 3,
    name: 'ジュニア',
    age: 1,
  },
];

export const AboutTab: FunctionComponent = () => {
  return (
    <>
      <AboutTabComponent data={data}></AboutTabComponent>
    </>
  );
};
