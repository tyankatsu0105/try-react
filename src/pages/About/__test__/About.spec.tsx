import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';

import { AboutTab } from '../components/AboutTab';

describe('Home', () => {
  it('render', () => {
    const { baseElement } = render(
      <MemoryRouter initialEntries={['/']}>
        <AboutTab />
      </MemoryRouter>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('text', () => {
    const { getByText } = render(
      <MemoryRouter initialEntries={['/']}>
        <AboutTab />
      </MemoryRouter>
    );
    expect(getByText('太郎')).toBeInTheDocument();
  });
});
