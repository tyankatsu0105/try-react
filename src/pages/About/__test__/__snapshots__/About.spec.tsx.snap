// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render 1`] = `
<body>
  <div>
    <a
      href="/about/1"
    >
      太郎
    </a>
    <a
      href="/about/2"
    >
      花子
    </a>
    <a
      href="/about/3"
    >
      ジュニア
    </a>
  </div>
</body>
`;
